dbms/src/Flash/Mpp/ExchangeReceiver.cpp

```diff
@@ -304,15 +304,31 @@ ExchangeReceiverBase<RPCContext>::ExchangeReceiverBase(
     , exc_log(Logger::get("ExchangeReceiver", req_id, executor_id))
     , collected(false)
 {
-    rpc_context->fillSchema(schema);
-    setUpConnection();
+    try
+    {
+        rpc_context->fillSchema(schema);
+        setUpConnection();
+    }
+    catch (...)
+    {
+        cancel();
+        thread_manager->wait();
```

Comment:
what if cancel() or thd_manager->wait() throw a exception, maybe need record first exception and also try {cancel & wait}